Test/git-issues/git-issue-1963b.dfy

```diff
@@ -0,0 +1,17 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/// This file checks that refinement is forbidden for datatypes
+
+module A {
+  datatype D = D
+  lemma P(d: D) ensures d == D {}
+}
+
+module B refines A {
+  datatype D = ... D | D'
```

Comment:
Same here. Add a comment that says this line gives an error.

Reply:
```suggestion
  datatype D = ... D | D' // Error: Cannot change (or repeat) constructors
```